socorro/processor/rules/breakpad.py

```diff
@@ -70,6 +70,70 @@ def action(self, raw_crash, dumps, processed_crash, status):
         )
 
 
+class TruncateStacksRule(Rule):
+    """Truncate stacks that are too large
+
+    This rule truncates stacks that are greater than MAX_FRAMES. It truncates the stacks
+    in the middle with the thinking that the beginning and end of the stack are the most
+    interesting.
+
+    """
+
+    MAX_FRAMES = 500
+    HALF_MAX_FRAMES = int(MAX_FRAMES / 2)
+
+    def __init__(self):
+        super().__init__()
+        self.metrics = markus.get_metrics("processor.truncatestacksrule")
+
+    def truncation_frame(self, truncated_frames):
+        return {"truncated": {"msg": f"{len(truncated_frames):,} frames truncated"}}
```

Comment:
I made this a structure in case we decide to "summarize" the truncated frames at some point in the future. By making it a structure, we can add additional information in a backwards-compatible way by adding new bits to the structure without changing existing bits.